packages/react-composites/tests/browser/chat/hermetic/Messaging.test.ts

```diff
@@ -21,4 +21,64 @@ test.describe('Tests related to messaging', async () => {
       'one-message-in-chat-thread.png'
     );
   });
+
+  test('Inline Image should show a placeholder box when loading', async ({ page, serverUrl }) => {
+    // Mock the api call before navigating
+    await page.route(serverUrl + '/images/inlineImageExample1.png', async (route) => {
+      setTimeout(async () => {
+        try {
+          await route.continue();
+        } catch (error) {
+          console.error('Failed at continue on route, Error: ', error);
+        }
+      }, 3000);
+    });
+
+    await waitForChatCompositeToLoad(page);
+
+    await page.goto(
+      buildUrlForChatAppUsingFakeAdapter(serverUrl, {
+        localParticipant: TEST_PARTICIPANTS[1],
+        remoteParticipants: [TEST_PARTICIPANTS[0], TEST_PARTICIPANTS[2]],
+        localParticipantPosition: 1,
+        sendRemoteInlineImageMessage: true
+      })
+    );
+
+    expect(await stableScreenshot(page, { stubMessageTimestamps: true })).toMatchSnapshot(
+      `messaging-inline-image-loading-stage.png`
+    );
+
+    await page.locator(dataUiId('SomeImageId1')).hover();
+
+    expect(await stableScreenshot(page, { stubMessageTimestamps: true })).toMatchSnapshot(
+      'messaging-inline-image-loaded-stage.png'
+    );
+  });
+
+  test('Inline Image should show a broken image icon when image fetch failed', async ({ page, serverUrl }) => {
+    // Mock the api call before navigating
+    await page.route(serverUrl + '/images/inlineImageExample1.png', async (route) => {
```

Comment:
Do we need timeout here?

Reply:
This test is for the network fail situation. We can mock a failure right away instead of waiting for a timeout. The image loading test needs the timeout to mimic the loading time.